src/tree/updater_gpu_hist.cu

```diff
@@ -676,6 +677,7 @@ struct DeviceShard {
   bst_uint row_begin_idx;
   bst_uint row_end_idx;
   bst_uint n_rows;
+  bst_uint rows_processed;
```

Comment:
Any way to do this without a member variable? Its difficult for someone looking at the shard object to see what this does, what state it carries and if it is necessary to the rest of the algorithm.

Reply:
done without altering the device shard state